src/Medidata.ZipkinTracer.Core.Collector/SpanProcessor.cs

```diff
@@ -15,33 +19,33 @@ public class SpanProcessor
         //send contents of queue if it has pending items but less than max batch size after doing max number of polls
         internal const int MAX_NUMBER_OF_POLLS = 5;
 
-        private TBinaryProtocol.Factory protocolFactory;
+        private readonly Uri uri;
         internal BlockingCollection<Span> spanQueue;
-        private IClientProvider clientProvider;
+        internal ConcurrentQueue<SerializableSpan> serializableSpans; 
 
-        internal List<LogEntry> logEntries;
         internal SpanProcessorTaskFactory spanProcessorTaskFactory;
         internal int subsequentPollCount;
         internal int retries;
         internal int maxBatchSize;
+        private readonly ILog logger;
 
-        public SpanProcessor(BlockingCollection<Span> spanQueue, IClientProvider clientProvider, int maxBatchSize, ILog logger)
+        public SpanProcessor(Uri uri, BlockingCollection<Span> spanQueue, int maxBatchSize, ILog logger)
         {
             if ( spanQueue == null) 
             {
                 throw new ArgumentNullException("spanQueue is null");
             }
 
-            if ( clientProvider == null) 
+            if (uri == null)
             {
-                throw new ArgumentNullException("clientProvider is null");
+                throw new ArgumentNullException("zipkin Uri is uri");
```

Comment:
The exception constructor argument can be just simply "uri" - as this argument is for the null paramater's name.